src/core/PathTracingSceneGenerator.js

```diff
@@ -33,7 +34,17 @@ export class PathTracingSceneGenerator {
 
 			}
 
-		} );
+		}
+
+		if ( scene instanceof Object3D ) {
+
+			scene.traverse( checkObject );
+
+		} else if ( scene instanceof Array ) {
+
+			scene.forEach( checkObject );
+
+		}
```

Comment:
And here I think it would be best if we just ensure the passed object is an array at the beginning of the function:

```js
prepScene( scene ) {

  scene = Array.isArray( scene ) ? scene : [ scene ];

  // ...
```

This should also simplify the function contents

Reply:
For both this and the previous comment, I think we might want to keep it the way it is in the PR.

We want different behavior when passing it an array vs a single scene object. For arrays, we want to filter each object individually, ignoring their children. For scenes, we want to traverse through the tree and filter all of the children.

If we cast everything to an array at the beginning of generation, we'd need to traverse through each object inside of checkObject to replicate the current behavior when passing a scene. Assuming we treat prefiltered arrays of objects the same way, we'll include all of their children in the output scene, even if they were filtered before being passed to the method.

In our case, we have characters with several control widgets/gizmos and geometries as their descendants. We don't want to display the gizmo geometries in the pathtracer. We filter them out before passing them to the pathtracer:

`
let filteredObjects = []
scene.traverse((c) => {
  if (!isGizmoGeometry(c)) {
    filteredObjects.push(c)
  }
})

const results = generator.generate(filteredObjects)
`

If we implement things the way you're suggesting, we'd still end up rendering those gizmos since their parents (the characters) are still in filteredObjects. You could suggest we make copies of the characters and remove their children before passing them to .generate, but at that point we might as well just copy the scene entirely and heavily modify it before passing it to the current version of .generate

Reply:
I think I feel strongly that passing a single scene in and an array of one scene in should behave the same. I think it would be confusing if these function calls behaved differently:

```js
generator.generate( scene );

generator.generate( [ scene ] );
```

Can you explain the structure of your hierarchy? Do your _meshes_ have children? Are you not using groups to store meshes and gizmos? Another option is to set the gizmos to be `visible = false` before updating the geometry and resetting them back to visible afterward.

Reply:
Ignore my prior comment - I think I misunderstood why some elements were making it pass my filtering function. Your way should work fine for our use case. PR will be updated shortly.